powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/controller/v3/ActivationController.java

```diff
@@ -79,7 +79,7 @@ public void setAuthenticationProvider(PowerAuthAuthenticationProvider authentica
     @RequestMapping(value = "create", method = RequestMethod.POST)
     @PowerAuthEncryption
     public ActivationLayer1Response createActivation(@EncryptedRequestBody ActivationLayer1Request request,
-                                                     PowerAuthEciesEncryption eciesEncryption) throws PowerAuthActivationException {
+                                                     PowerAuthEciesEncryption<ActivationLayer1Request> eciesEncryption) throws PowerAuthActivationException {
```

Comment:
The problem is actually this. We don't want to have any additional `PowerAuthEciesEncryption<ActivationLayer1Request> eciesEncryption` parameter...

Reply:
As agreed, we will switch to detecting the request type using `@EncryptedRequestBody` annotation marking the request object and remove the unnecessary type.